src/AspNet.Security.OAuth.SuperOffice/Implementation/DefaultSuperOfficeIdTokenValidator.cs

```diff
@@ -36,33 +35,34 @@ public override async Task ValidateAsync([NotNull] SuperOfficeValidateIdTokenCon
                 throw new NotSupportedException($"The configured {nameof(JwtSecurityTokenHandler)} cannot validate tokens.");
             }
 
-            var keySet = await _configManager.GetPublicKeySetAsync(context);
-
-            var parameters = new TokenValidationParameters()
+            if (context.Options.ConfigurationManager != null)
             {
-                ValidAudience = context.Options.ClientId,
-                ValidIssuer = context.Options.ClaimsIssuer,
-                IssuerSigningKeys = keySet.Keys,
-            };
+                var openIdConnectConfiguration = await context.Options.ConfigurationManager.GetConfigurationAsync(CancellationToken.None);
+                context.Options.TokenValidationParameters.IssuerSigningKeys = openIdConnectConfiguration.JsonWebKeySet.Keys;
 
-            try
-            {
-                _tokenHandler.ValidateToken(context.IdToken, parameters, out var _);
-            }
-            catch (Exception ex)
-            {
-                _logger.LogError(
-                    ex,
-                    "SuperOffice ID token validation failed for issuer {TokenIssuer} and audience {TokenAudience}.",
-                    parameters.ValidIssuer,
-                    parameters.ValidAudience);
+                try
+                {
+                    _tokenHandler.ValidateToken(context.IdToken, context.Options.TokenValidationParameters, out var _);
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(
+                        ex,
+                        "SuperOffice ID token validation failed for issuer {TokenIssuer} and audience {TokenAudience}.",
+                        context.Options.TokenValidationParameters.ValidIssuer,
+                        context.Options.TokenValidationParameters.ValidAudience);
 
-                _logger.LogTrace(
-                    ex,
-                    "SuperOffice ID token {IdToken} could not be validated.",
-                    context.IdToken);
+                    _logger.LogTrace(
+                        ex,
+                        "SuperOffice ID token {IdToken} could not be validated.",
+                        context.IdToken);
 
-                throw;
+                    throw;
+                }
+            }
+            else
+            {
+                throw new InvalidOperationException($"The ConfigurationManager is null.");
```

Comment:
The code would be cleaner if you did `if (context.Options.ConfigurationManager == null)` up top, as then you don't need the `else` block and the indenting changes.

Reply:
I don't think this message would be obvious to debug for someone integrating the provider who wasn't familiar with the implementation. Maybe something like `$"An OpenID Connect configuration manager has not been set on the {nameof(SuperOfficeAuthenticationOptions)} instance."`

Reply:
Good suggestion! I'll update that.